internal/pkg/manifest/validate.go

```diff
@@ -1498,6 +1498,15 @@ func (s SidecarConfig) validate() error {
 	}
 	return s.ImageOverride.validate()
 }
+func (s SidecarConfig) validateSideCarImageFields() error {
+	if s.Image.Basic == nil && s.Image.Advanced.Location == nil && s.Image.Advanced.Build.isEmpty() {
+		return fmt.Errorf(`must specify one of image or image.build or image.location`)
+	}
+	if err := s.Image.Advanced.validate(); err != nil {
```

Comment:
this will call validate on the correct underlying type (Advanced, in this case, if it's set)
```suggestion
	if err := s.Image.validate(); err != nil {
```